src/fs.rs

```diff
@@ -166,18 +187,22 @@ fn symlink_attr(ino: u64) -> FileAttr {
     }
 }
 
-fn _which<P>(path: &Path, exe_name: P) -> Option<PathBuf>
+fn _which<P1, P2>(path: &Path, exe_name: P1, mountpoints: &[P2]) -> Option<PathBuf>
 where
-    P: AsRef<Path>,
+    P1: AsRef<Path>,
+    P2: AsRef<Path>,
 {
-    let skip_path = match path.symlink_metadata() {
-        Ok(stat) => stat.nlink() as u32 == ENVFS_MAGIC,
-        Err(_) => true,
-    };
-    if skip_path {
+    if mountpoints.iter().any(|m| m.as_ref().starts_with(path)) {
         return None;
     }
 
+    // Do we still need this check if we already check for mountpoints?
+    if let Ok(stat) = path.symlink_metadata() {
+        if stat.nlink() as u32 == ENVFS_MAGIC {
+            return None;
+        }
+    }
```

Comment:
If someone bind-mounts say `/bin/` to somewhere else and adds it to the `PATH`, this would deadlock again. Not sure what happens if this is removed. Infinite recursion or deadlock somewhere else?

Reply:
Yes. You are no longer allowed to manually bind mount envfs. All mountpoints needs to be known to envfs through the bind-mount option. And yes, if we don't have this check it does infinite recursion